userspace/sysdig/chisels/spy_file.lua

```diff
@@ -93,7 +103,23 @@ function on_init()
 		filter = string.format("%s%s", filter, "evt.is_io=true and ")
 	end
 
-	filter = string.format("%s%s", filter, "fd.type=file and evt.dir=< and evt.failed=false")
+	if type_of_file == "F" or type_of_file == "f" then
+		filter = string.format("%s%s", filter, "fd.type=file and ")
+	elseif type_of_file == "P" or type_of_file == "p" then
+		filter = string.format("%s%s", filter, "fd.type=pipe and ")
+	elseif type_of_file == "U" or type_of_file == "u" then
+		filter = string.format("%s%s", filter, "fd.type=unix and ")
+	elseif type_of_file == "FP" or type_of_file == "fP" or type_of_file == "Fp" or type_of_file == "fp" or type_of_file == "PF" or type_of_file == "pF" or type_of_file == "Pf" or type_of_file == "pf" then
+		filter = string.format("%s%s", filter, "(fd.type=file or fd.type=pipe) and ")
+	elseif type_of_file == "FU" or type_of_file == "fU" or type_of_file == "Fu" or type_of_file == "fu" or type_of_file == "UF" or type_of_file == "uF" or type_of_file == "Uf" or type_of_file == "uf" then
+		filter = string.format("%s%s", filter, "(fd.type=file or fd.type=unix) and ")
+	elseif type_of_file == "PU" or type_of_file == "pU" or type_of_file == "Pu" or type_of_file == "pu" or type_of_file == "UP" or type_of_file == "uP" or type_of_file == "Up" or type_of_file == "up" then
+		filter = string.format("%s%s", filter, "(fd.type=pipe or fd.type=unix) and ")
+	else
+		filter = string.format("%s%s", filter, "(fd.type=file or fd.type=pipe or fd.type=unix) and ")
```

Comment:
For "backward compatibility" maybe I'd keep just fd.type=file by default. There are already other chisels that intercept everything (`echo_fds` for example).